src/hotspot/share/gc/g1/heapRegionRemSet.hpp

```diff
@@ -150,6 +150,7 @@ class HeapRegionRemSet : public CHeapObj<mtGC> {
   // the heap region that owns this RSet.
   void add_code_root(nmethod* nm);
   void remove_code_root(nmethod* nm);
+  void remove_unlinked_nmethods();
```

Comment:
Instead of introducing new naming scheme with `remove_unlinked_` I suggest we add a variation of `unregister_nmethods` such as `bulk_unregister_nmethods` so that the relation with `register_nmethod` and associated Closures is maintained. Another example is `RegisterNMethodOopClosure`  vs `RemoveUnlinkedHeapRegionClosure`.